### PR TITLE
Feature/aut 118 tests

### LINKED ISSUE #12: Tech Support Module
Demonstrate that error handling is working properly 
Demonstrate that logging is working properly 
Check the notifications mechanism 

See the tech_support*.* spec documents for details.

### PR DESCRIPTION
**Pull request by rclanan**
Refactoring of how tests were being setup and handled.
Removed dependency of extra karma plugins that were no longer needed.
Resovled outstanding issues with tests
Added ability to debug tests using npm run test:debug

Closes #12

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -2,15 +2,14 @@ var path = require('path');
...
 
-var unitTestEntry = 'src/app/specs.webpack.js';
+var unitTestEntry = 'test/unit/helper.js';
 
 // run multiple times in watch mode
 var singleRun = !args.watch;
-
 // use phantomjs in watch mode
-var browser = 'PhantomJS';
+var browser = args.debug ? ['Chrome'] : ['PhantomJS'];
```
Comment (BWortman) [inline karma.conf.js]: The readme should be updated to explain how to access this functionality, and to list Chrome as a prerequisite for using it.

File: karma.conf.js [hunk 2]
```diff
@@ -73,16 +78,21 @@ module.exports = function karmaConfig(config) {
...
-      }
+      },
+      noInfo : true
     },
     preprocessors: processors,
     reporters: reporters,
     coverageReporter: {
       dir: './coverage',
-      reporters: coverageReporters
+      reporters: coverageReporters,
```
Comment (BWortman) [inline karma.conf.js]: Should add usage of this to the readme.

File: src/app/data-access/raise-error-data-service-factory.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+dataAccess.$inject = [
+  '$q',
+  '$http',
+  'generalConfig',
+  'strings'
+];
+
+dataAccess
+  .factory('raiseErrorDataService', raiseErrorDataService);
```
Comment (BWortman) [inline src/app/data-access/raise-error-data-service-factory.js]: Need to prefix the factory name with "fa".

File: src/app/data-access/root-data-service-factory.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+dataAccess.$inject = [
+  '$http',
+  'generalConfig',
+  'strings'
+];
+
+dataAccess
+  .factory('rootDataService', rootDataService);
```
Comment (BWortman) [inline src/app/data-access/root-data-service-factory.js]: Need to prefix the factory name with "fa".

File: src/app/tech-support/tech-support.controller.spec.js [hunk 5]
```diff
@@ -1,8 +1,153 @@
...
+
+  beforeEach(angular.mock.module('fa.tech-support'));
+
+  angular.mock.module(function($provide){
+    $provide.value('rootDataService', inputs.stubs.rootDataService);
+    $provide.value('raiseErrorDataService', inputs.stubs.raiseErrorDataService);
+  });
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
```
Comment (BWortman) [inline src/app/tech-support/tech-support.controller.spec.js]: You aren't using the sandbox anywhere. I deleted everything related to the sandbox variable and all tests still passed.

File: src/app/tech-support/tech-support.controller.spec.js [hunk 6]
```diff
@@ -1,8 +1,153 @@
...
+
+    angular.mock.inject(function ($injector) {
```
Comment (BWortman) [inline src/app/tech-support/tech-support.controller.spec.js]: Why make this different than every other controller test? We don't using $injector in this way for the others. Should be consistent.

File: src/app/tech-support/tech-support.controller.spec.js [hunk 7]
```diff
@@ -1,8 +1,153 @@
...
+
+
+  describe('checkHttpErrorHandlerError', function checkHttpErrorHandlerErrorTest() {
+    it('should log unexpected success', function testImpl() {
+      controller.checkHttpErrorHandlerError();
+
+      $scope.$apply();
+
+      inputs.stubs.raiseErrorDataService.getRaiseError.should.have.been.called;
+    });
```
Comment (BWortman) [inline src/app/tech-support/tech-support.controller.spec.js]: Should also verify that the $log method was called with the correct args.

File: karma.conf.js (hunk 1 above)
Comment (rclanan) [inline karma.conf.js]: This should be taken care of.

File: karma.conf.js (hunk 2 above)
Comment (rclanan) [inline karma.conf.js]: Not sure what is meant by this. This is used by coveralls. We can add a badge to the readme for coveralls.

File: src/app/data-access/raise-error-data-service-factory.js (hunk 3 above)
Comment (rclanan) [inline src/app/data-access/raise-error-data-service-factory.js]: Done!

File: src/app/data-access/root-data-service-factory.js (hunk 4 above)
Comment (rclanan) [inline src/app/data-access/root-data-service-factory.js]: Done!

File: src/app/tech-support/tech-support.controller.spec.js (hunk 5 above)
Comment (rclanan) [inline src/app/tech-support/tech-support.controller.spec.js]: Removed

File: src/app/tech-support/tech-support.controller.spec.js (hunk 7 above)
Comment (rclanan) [inline src/app/tech-support/tech-support.controller.spec.js]: Done

**Pull request merged**